File: src/main/java/net/ccbluex/liquidbounce/ui/gui/menu/GuiCommitInfo.java
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge by LiquidBounce.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.ui.gui.menu;

import me.zywl.fdpclient.FDPClient;
import net.ccbluex.liquidbounce.ui.font.Fonts;
import net.ccbluex.liquidbounce.utils.GitUtils;
import net.ccbluex.liquidbounce.utils.render.RenderUtils;
import net.minecraft.client.gui.GuiButton;
import net.minecraft.client.gui.GuiScreen;
import net.minecraft.util.ResourceLocation;

import java.awt.*;
import java.io.IOException;

public class GuiCommitInfo extends GuiScreen {
    public static final ResourceLocation gitImage = new ResourceLocation("fdpclient/gui/icons/git.png");

    /**
     * Adds the buttons (and other controls) to the screen in question. Called when the GUI is displayed and when the
     * window resizes, the buttonList is cleared beforehand.
     */
    @Override
    public void initGui() {
        this.buttonList.add(new GuiButton(0, 70, 30 + Fonts.minecraftFont.FONT_HEIGHT * 7 + 20, "Back"));
        super.initGui();
    }

    @Override
    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
        super.drawDefaultBackground();
        RenderUtils.drawImage(gitImage, 30, 30, 30, 30);
        String buildUser = GitUtils.gitInfo.getProperty("git.build.user.name");
        String version = GitUtils.gitInfo.getProperty("git.build.version");
        String commitId = GitUtils.gitInfo.getProperty("git.commit.id");
        String commitIdAbbrev = GitUtils.gitInfo.getProperty("git.commit.id.abbrev");
        String commitMessage = GitUtils.gitInfo.getProperty("git.commit.message.short");
        String branch = GitUtils.gitInfo.getProperty("git.branch");
        String repo = GitUtils.gitInfo.getProperty("git.remote.origin.url");
        this.drawString(Fonts.minecraftFont, "Git Info", 70 ,30, new Color(255, 255, 255).getRGB());
        this.drawString(Fonts.minecraftFont, FDPClient.CLIENT_NAME + " built by " + buildUser, 70, 30 + Fonts.minecraftFont.FONT_HEIGHT, new Color(255, 255, 255).getRGB());
        this.drawString(Fonts.minecraftFont, "Version: " + version, 70, 30 + Fonts.minecraftFont.FONT_HEIGHT * 2 + 5, new Color(255, 255, 255).getRGB());
        this.drawString(Fonts.minecraftFont, "CommitId: " + commitId + " (" + commitIdAbbrev + ")", 70, 30 + Fonts.minecraftFont.FONT_HEIGHT * 3 + 5, new Color(255, 255, 255).getRGB());
        this.drawString(Fonts.minecraftFont, "CommitMessage: " + commitMessage, 70, 30 + Fonts.minecraftFont.FONT_HEIGHT * 4 + 5, new Color(255, 255, 255).getRGB());
        this.drawString(Fonts.minecraftFont, "Branch: " + branch, 70, 30 + Fonts.minecraftFont.FONT_HEIGHT * 5 + 5, new Color(255, 255, 255).getRGB());
        this.drawString(Fonts.minecraftFont, "Remote origin: " + repo, 70, 30 + Fonts.minecraftFont.FONT_HEIGHT * 6 + 5, new Color(255, 255, 255).getRGB());
        this.drawString(Fonts.minecraftFont, "Developers: " + String.join(" ", FDPClient.CLIENT_DEV), 70, 30 + Fonts.minecraftFont.FONT_HEIGHT * 7 + 5, new Color(255, 255, 255).getRGB());
        super.drawScreen(mouseX, mouseY, partialTicks);
    }

    @Override
    protected void actionPerformed(GuiButton button) throws IOException {
        if (button.id == 0) {
            mc.displayGuiScreen(null);
        }
    }
}

File: src/main/java/net/ccbluex/liquidbounce/ui/client/gui/GuiCommitInfo.kt
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.ui.client.gui

import net.ccbluex.liquidbounce.FDPClient
import net.ccbluex.liquidbounce.FDPClient.CLIENT_NAME
import net.ccbluex.liquidbounce.features.module.modules.client.HUDModule.guiColor
import net.ccbluex.liquidbounce.handler.api.ClientUpdate
import net.ccbluex.liquidbounce.ui.font.AWTFontRenderer.Companion.assumeNonVolatile
import net.ccbluex.liquidbounce.ui.font.Fonts
import net.ccbluex.liquidbounce.utils.render.RenderUtils.drawBloom
import net.ccbluex.liquidbounce.utils.render.RenderUtils.drawImage
import net.ccbluex.liquidbounce.utils.ui.AbstractScreen
import net.minecraft.client.gui.GuiButton
import net.minecraft.util.ResourceLocation
import java.awt.Color
import java.io.IOException
import kotlin.Float
import kotlin.Int
import kotlin.Throws

class GuiCommitInfo : AbstractScreen() {

    private val gitImage: ResourceLocation = ResourceLocation("${CLIENT_NAME.lowercase()}/texture/mainmenu/github.png")

    override fun initGui() {
        val buttonWidth = 200
        val buttonHeight = 20
        val buttonX = width / 2 - buttonWidth / 2
        val buttonY = height - buttonHeight - 10

        +GuiButton(0, buttonX, buttonY, buttonWidth, buttonHeight, "Back")

        super.initGui()
    }

    override fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float) {

        assumeNonVolatile = true

        drawDefaultBackground()
        drawImage(gitImage, 30, 30, 30, 30)

        val font = Fonts.minecraftFont
        val textColor = Color(255, 255, 255).rgb
        val startX = 70
        val startY = 30

        val lines = listOf(
            "Git Info",
            "$CLIENT_NAME built by ${ClientUpdate.gitInfo.getProperty("git.build.user.name")}",
            "Version: ${ClientUpdate.gitInfo.getProperty("git.build.version")}",
            "CommitId: ${ClientUpdate.gitInfo.getProperty("git.commit.id")} (${ClientUpdate.gitInfo.getProperty("git.commit.id.abbrev")})",
            "CommitMessage: ${ClientUpdate.gitInfo.getProperty("git.commit.message.short")}",
            "Branch: ${ClientUpdate.gitInfo.getProperty("git.branch")}",
            "Remote origin: ${ClientUpdate.gitInfo.getProperty("git.remote.origin.url")}",
            "Developers: ${FDPClient.CLIENT_AUTHOR}"
        )

        lines.forEachIndexed { index, line ->
            drawString(font, line, startX, startY + font.FONT_HEIGHT * index + index * 5, textColor)
        }

        drawBloom(mouseX - 5, mouseY - 5, 10, 10, 16, Color(guiColor))

        assumeNonVolatile = false

        super.drawScreen(mouseX, mouseY, partialTicks)
    }

    @Throws(IOException::class)
    override fun actionPerformed(button: GuiButton) {
        if (button.id == 0) {
            mc.displayGuiScreen(null)
        }
    }
}